src/utils.ts

```diff
@@ -29,44 +29,84 @@ function hasScope(file: string) {
   return file.match(/^@/);
 }
 
-export async function findPackages(storagePath: string, validationHandler: Function) {
-  const listPackages: Array<any> = [];
+async function getVersions(scopePath: string, packageName: string): Promise<string[]> {
+  const versions: string[] = [];
+  const arr: string[] = await readDirectory(scopePath);
+  arr.forEach( (filePath: string) => {
```

Comment:
I would suggest to use `map` or `reduce`.

Reply:
Fixed: cannot imagine how to use reduce here, but I changed it to use `filter` and `map`.